src/Container.php

```diff
@@ -19,7 +19,7 @@ final class Container
 {
 
     /**
-     * @var Container
+     * @var self|null
```

Comment:
This class is final and extends nothing, so and instance of it can only be an instance of it. In such cases, I prefer to be explicit in declaring the type, so I prefer `Container` over `self`